diff --git a/swift/common/middleware/tempauth.py b/swift/common/middleware/tempauth.py
index 28d8251db..41aee6c82 100644
--- a/swift/common/middleware/tempauth.py
+++ b/swift/common/middleware/tempauth.py
@@ -275,7 +275,7 @@ class TempAuth(object):
             return self.app(env, start_response)
         if env.get('PATH_INFO', '').startswith(self.auth_prefix):
             return self.handle(env, start_response)
-        s3 = env.get('s3api.auth_details')
+        s3 = env.get('s3api.auth_details') or env.get('swift3.auth_details')
         token = env.get('HTTP_X_AUTH_TOKEN', env.get('HTTP_X_STORAGE_TOKEN'))
         service_token = env.get('HTTP_X_SERVICE_TOKEN')
         if s3 or (token and token.startswith(self.reseller_prefix)):
@@ -441,7 +441,8 @@ class TempAuth(object):
             else:
                 groups = groups.encode('utf8')

-        s3_auth_details = env.get('s3api.auth_details')
+        s3_auth_details = env.get('s3api.auth_details') or\
+            env.get('swift3.auth_details')
         if s3_auth_details:
             if 'check_signature' not in s3_auth_details:
                 self.logger.warning(
